pallets/api/src/nonfungibles/mod.rs

```diff
@@ -0,0 +1,668 @@
+		#[pallet::call_index(7)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::mint())]
+		pub fn mint(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			to: AccountIdOf<T>,
+			item: ItemIdOf<T>,
+			witness: MintWitness<ItemIdOf<T>, ItemPriceOf<T>>,
+		) -> DispatchResult {
+			NftsOf::<T>::mint(
+				origin,
+				collection,
+				item,
+				T::Lookup::unlookup(to.clone()),
+				Some(witness),
+			)?;
+			Self::deposit_event(Event::Transfer { collection, item, from: None, to: Some(to) });
+			Ok(())
+		}
+
+		/// Destroys the specified item. Clearing the corresponding approvals.
+		///
+		/// # Parameters
+		/// - `collection` - The collection.
+		/// - `item` - The item to burn.
+		#[pallet::call_index(8)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::burn())]
+		pub fn burn(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: ItemIdOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let owner =
+				NftsOf::<T>::owner(collection, item).ok_or(NftsErrorOf::<T>::UnknownItem)?;
+			NftsOf::<T>::burn(origin, collection, item)?;
+			Self::deposit_event(Event::Transfer { collection, item, from: Some(owner), to: None });
+			Ok(().into())
+		}
+
+		/// Creates an NFT collection.
+		///
+		/// # Parameters
+		/// - `admin` - The admin account of the collection.
+		/// - `config` - Settings and config to be set for the new collection.
+		#[pallet::call_index(12)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::create())]
+		pub fn create(
+			origin: OriginFor<T>,
+			admin: AccountIdOf<T>,
+			config: CollectionConfigOf<T>,
+		) -> DispatchResult {
+			let creator = ensure_signed(origin.clone())?;
+			let id = NextCollectionIdOf::<T>::get()
+				.or(T::CollectionId::initial_value())
+				.ok_or(NftsErrorOf::<T>::UnknownCollection)?;
+			NftsOf::<T>::create(origin, T::Lookup::unlookup(admin.clone()), config)?;
+			Self::deposit_event(Event::Created { id, creator, admin });
+			Ok(())
+		}
+
+		/// Destroy an NFT collection.
+		///
+		/// # Parameters
+		/// - `collection` - The collection to be destroyed.
+		/// - `witness` - Information on the items minted in the `collection`. This must be
+		/// correct.
+		#[pallet::call_index(13)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::destroy(
+    		witness.item_metadatas,
+    		witness.item_configs,
+    		witness.attributes,
+		))]
+		pub fn destroy(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			witness: DestroyWitness,
+		) -> DispatchResultWithPostInfo {
+			NftsOf::<T>::destroy(origin, collection, witness)
+		}
+
+		/// Set an attribute for a collection or item.
+		///
+		/// Origin must be Signed and must conform to the namespace ruleset:
+		/// - `CollectionOwner` namespace could be modified by the `collection` Admin only;
+		/// - `ItemOwner` namespace could be modified by the `item` owner only. `item` should be set
+		///   in that case;
+		/// - `Account(AccountId)` namespace could be modified only when the `origin` was given a
+		///   permission to do so;
```

Comment:
Perhaps also good to add. This validates the point made earlier to do a final check with pallet nfts if there is any docs we should add to the dispatchable docs here and forgotten in the spec.
```suggestion
		///   permission to do so;
		///
		/// The funds of `origin` are reserved according to the formula:
		/// `AttributeDepositBase + DepositPerByte * (key.len + value.len)` taking into
		/// account any already reserved funds.
```